server/src/main/scala/org/http4s/armeria/server/ArmeriaServerBuilder.scala

```diff
@@ -38,29 +40,31 @@ import org.http4s.server.{
 import org.http4s.server.defaults.{IdleTimeout, ResponseTimeout, ShutdownTimeout}
 import org.http4s.syntax.all._
 import org.log4s.{Logger, getLogger}
+
 import scala.collection.immutable
 import scala.concurrent.duration.FiniteDuration
 
 sealed class ArmeriaServerBuilder[F[_]] private (
     armeriaServerBuilder: ArmeriaBuilder,
     socketAddress: InetSocketAddress,
     serviceErrorHandler: ServiceErrorHandler[F],
-    banner: List[String]
-)(implicit protected val F: ConcurrentEffect[F])
+    banner: List[String],
+    dispatcher: Dispatcher[F]
+)(implicit protected val F: Async[F])
     extends ServerBuilder[F] {
   override type Self = ArmeriaServerBuilder[F]
 
-  private[this] val logger: Logger = getLogger
-
   type DecoratingFunction = (HttpService, ServiceRequestContext, HttpRequest) => HttpResponse
 
+  private[this] val logger: Logger = getLogger
+
   override def bindSocketAddress(socketAddress: InetSocketAddress): Self =
     copy(socketAddress = socketAddress)
 
   override def withServiceErrorHandler(serviceErrorHandler: ServiceErrorHandler[F]): Self =
     copy(serviceErrorHandler = serviceErrorHandler)
 
-  override def resource: Resource[F, ArmeriaServer[F]] =
```

Comment:
`withTls` builder method on line 257 seems not to compile with Scala 2.12.
Could you use `both()` instead of `tupled`?
```scala
def withTls(...): F[Self] = 
    keyCertChainInputStream
	    .both(keyInputStream)
	    .use { case (x, y) =>
	      F.delay {
	        armeriaServerBuilder.tls(x, y, keyPassword.orNull)
	        this
	      }
      
```